src/modules/75-ce/components/COGatewayList/COGatewayList.tsx

```diff
@@ -468,8 +470,63 @@ interface RulesTableContainerProps {
   searchParams: SearchParams
 }
 
+const POLL_TIMER = 1000 * 60 * 1
+
+const useSubmittedRulesStatusUpdate = ({
+  rules,
+  onRuleUpdate
+}: {
+  rules: Service[]
+  onRuleUpdate?: (params: { updatedService: Service; index: number }) => void
+}) => {
+  const { accountId } = useParams<AccountPathProps>()
+  const { data, refetch, loading } = useRouteDetails({ account_id: accountId, rule_id: 0, lazy: true })
+  const rulesToFetch = useRef<{ index: number; rule: Service }[]>([])
+  const timer = useRef<NodeJS.Timer | null>(null)
+
+  const clearTimer = () => {
+    clearTimeout(timer.current as NodeJS.Timer)
+    timer.current = null
+  }
+
+  useEffect(() => {
+    rulesToFetch.current = []
+    clearTimer()
+    rules.forEach((r, i) => {
+      if (r.status === 'submitted') {
+        rulesToFetch.current.push({ index: i, rule: r })
+      }
+    })
+  }, [rules])
+
+  const triggerRuleFetching = () => {
+    timer.current = setTimeout(() => {
+      refetch({ pathParams: { account_id: accountId, rule_id: rulesToFetch.current[0].rule.id } })
+      clearTimer()
+    }, POLL_TIMER)
+  }
+
+  useEffect(() => {
+    if (!_isEmpty(rulesToFetch.current) && !loading && timer.current === null) {
+      triggerRuleFetching()
+    }
+  }, [rulesToFetch.current, timer.current])
+
+  useEffect(() => {
+    if (!_isEmpty(data?.response) && data?.response?.service?.status !== 'submitted') {
+      onRuleUpdate?.({ updatedService: data?.response?.service as Service, index: rulesToFetch.current[0].index })
+      rulesToFetch.current.shift()
+      clearTimer()
+    }
+    if (timer.current) {
+      return () => clearTimeout(timer.current as NodeJS.Timer)
+    }
+  }, [data?.response])
```

Comment:
Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'onRuleUpdate'. Either include it or remove the dependency array. If 'onRuleUpdate' changes too often, find the parent component that defines it and wrap that definition in useCallback.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232246)
```suggestion
  }, [data?.response, onRuleUpdate])
```